### DISCUSSION TITLE
Addition of endpoint to mailchimp plugin and refactor to typescript

### DISCUSSION THREAD

**Discussion by tomekpur** [medusajs/medusa#3985] [Feature Requests]
The current mailchimp plugin is quite limited in it's features, even with a single endpoint that is provided in the plugin. The current POST /mailchimp/subscribe only accepts new subscribers (by mailchimp api design).

To overcome the issue where an emailadress can only be added once, there is another endpoint available in the mailchimp api (with a PUT instead of a POST). I've been working on an update of this plugin to add the endpoint PUT /mailchimp/subscribe.

Whilst doing that, I'd like to refactor the code to use the newest utils and upgrade to typescript.

Any thoughts?

**Comment by srindom** [comment]
We would love to accept a PR on that 👍

**Reply by tomekpur** [reply]
There you go 😉 
[#4040](https://github.com/medusajs/medusa/pull/4040)
